src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
+using Microsoft.NET.TestFramework;
+using Xunit.Abstractions;
+using NuGet.Configuration;
+
+namespace Microsoft.NET.TestFramework
+{
+    public abstract class AspNetSdkTest : SdkTest
+    {
+        private readonly IEnumerable<System.Reflection.AssemblyMetadataAttribute> TestAssemblyMetadata;
+        public readonly string DefaultTfm;
+
+        private List<string> NuGetFeedsRetained = new List<string>(){"dotnet-public", "dotnet6", "dotnet6-transport"};
+
+        protected AspNetSdkTest(ITestOutputHelper log) : base(log) {
+            var assembly = Assembly.GetCallingAssembly();
+            TestAssemblyMetadata = assembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+            DefaultTfm = TestAssemblyMetadata.SingleOrDefault(a => a.Key == "AspNetTestTfm").Value;
+        }
+
+        private void CopyAndModifyNugetConfig(string projectDirectory)
+        {
+            // Copy the config at the root of the repo to `projectDirectory`
+            var repoRoot = TestAssemblyMetadata.SingleOrDefault(a => a.Key == "ArtifactsTmpDir").Value;
+            var configAtRoot = Path.Combine(repoRoot, "nuget.config");
```

Comment:
Does this fail on case-sensitive oses? The file in the root has a very specific case: https://github.com/dotnet/sdk/blob/master/NuGet.config (Btw, we could fix that up)